components/book/Form.tsx

```diff
@@ -5,37 +5,45 @@ import { Book } from "../../interfaces/Book";
 import Link from "next/link";
 
 interface Props {
-	book: Book;
+	book?: Book;
 }
 
 export const Form: FunctionComponent<Props> = ({ book }) => {
-	const handleDelete = () => {
+
+  const handleDelete = () => {
 		if (window.confirm("Are you sure you want to delete this item ?")) {
 			try {
 				fetch(`${book["@id"]}`, { method: "delete" });
 			} catch (error) {
-				alert("Error when delete element");
+				alert("error when delete element");
 				console.error(error);
 			}
 		}
 	};
 
-	const defaultValues = book["@id"]
-		? {
-				isbn: "",
-				title: "",
-				description: "",
-				author: "",
-				publicationDate: "",
-				reviews: "",
-		  }
-		: { ...book };
+	const defaultValues = () => {
+    console.log("book", book)
+    if (book["@id"] !== undefined) {
+      return
+      ({ ...book })
+    }
+    else {
+      return ({
```

Comment:
To prevent this, maybe could we transform the `Book` interface in a class, and use `new`. WDYT @gregcop1 @soyuka?

So we could just do: `<Formik initialValues={book ?? new Book()} ...`